lib/loggly/common.js

```diff
@@ -188,16 +188,10 @@ common.loggly = function () {
   }
   function sendInputLogs(input) {
     try {
-      request(requestOptions, function (err, res, body) {
-        if (err) {
-          // In rare cases server is busy
-          if (err.code === 'ETIMEDOUT' || err.code === 'ECONNRESET' || err.code === 'ESOCKETTIMEDOUT' || err.code === 'ECONNABORTED') {
-            retryOnError(input, err);
-          } else {
-            return onError(err);
-          }
-        } else {
-            responseCode = res.statusCode;
+      fetch(uri, requestOptions)
+        .then(function (res) {
+          return res.json().then(function (json) {
```

Comment:
Smells like callback hell. I wanted to suggest
```
fetch(uri, requestOptions)
  .then(function (res) { return res.json() })
  .then(function (json) {
```
but you do need `res` lower in the code, which makes that awkward. I guess you could store the `res` into a variable in the parent scope, that still seems less ugly than this nesting. I'm tempted to suggest to just rewrite this all to async await and arrow function since both are supported since node 7.10...